Remora.Rest/Json/Internal/BoundDataObjectConverter.cs

```diff
@@ -0,0 +1,272 @@
+        _allowExtraProperties = source.DoesAllowExtraProperties;
+
+        var dtoProperties = source.DTOProperties;
+
+        var writeProperties = new List<DTOPropertyInfo>();
+        var readProperties = new List<DTOPropertyInfo>();
+
+        var interfaceMap = typeof(TImplementation).GetInterfaceMap(typeof(TInterface));
+
+        foreach (var property in dtoProperties)
+        {
+            var converter = source.GetConverter(property, options);
+            var propertyOptions = GetPropertyConverterOptions(options, converter);
+            var defaultValue = source.GetDefaultValueForType(property.PropertyType);
+            var readNames = source.GetReadJsonPropertyName(property, options);
+            var writeNames = source.GetWriteJsonPropertyName(property, options);
+            var write = source.GetPropertyWriter(property);
+
+            // We cache this as well since the check is somewhat complex
+            bool allowsNull = property.AllowsNull();
+
+            var data = new DTOPropertyInfo
+            (
+                property,
+                readNames,
+                writeNames,
+                write,
+                allowsNull,
+                defaultValue,
+                propertyOptions,
+                readProperties.Count
+            );
+
+            if (property.CanWrite)
+            {
+                // If a property is writable, it can be *read* from JSON.
+                readProperties.Add(data);
+            }
+
+            if (property.CanWrite || source.IncludesReadOnlyProperty(property))
+            {
+                // Any property that is writable and not excluded due to being read-only,
+                // can be *written* to JSON.
+                writeProperties.Add(data);
+            }
+        }
+
+        _writeProperties = writeProperties.ToArray();
+        _readProperties = readProperties.ToArray();
+
+        _readPropertiesByName = readProperties
+            .SelectMany(p => p.ReadNames.Select((n, i) => (IsPrimary: i == 0, n, DTOProperty: p)))
+            .ToDictionary(x => x.n, x => (x.IsPrimary, x.DTOProperty));
+    }
+
+    private static JsonSerializerOptions GetPropertyConverterOptions(JsonSerializerOptions options, JsonConverter? converter)
+    {
+        if (converter == null)
+        {
+            return options;
+        }
+
+        var cloned = new JsonSerializerOptions(options);
+        cloned.Converters.Insert(0, converter);
+
+        return cloned;
+    }
+
+    /// <inheritdoc />
+    public override TInterface Read
+    (
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options
+    )
+    {
+        if (reader.TokenType != JsonTokenType.StartObject ||
+            !reader.Read())
+        {
+            DataObjectConverterShared.ThrowJsonException();
+        }
+
+        var readProperties = _readProperties;
+
+        // CMBK: Considering to cache this array and reuse it.
```

Comment:
What does "CMBK" mean?

Reply:
I'll rephrase it. Should indicate "Come back" since this is potentially an optimization. I just wasn't sure about it, and it seems neither are you.

Reply:
https://github.com/Remora/Remora.Rest/blob/37c3d450f268a39b3baae9976a071a09f2a5be7b/Remora.Rest/Json/Internal/BoundDataObjectConverter.cs#L153-L154
Considering the other point about not doing this optimization for now, should I remove the comment or leave it be?